File: testdata/bench/rust/BenchmarkTest00082.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Safe: XSS prevented with ammonia HTML sanitizer
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("name").unwrap_or(&String::new()).clone();
    let clean = ammonia::clean(&param);
    let html = format!("<html><body>Hello, {}!</body></html>", clean);
    HttpResponse::Ok().content_type("text/html").body(html)
}

File: testdata/bench/cpp/BenchmarkTest00082.cpp
#include <iostream>
#include <cstdlib>

int main() {
    std::cout << "Content-Type: text/html\r\n\r\n";
    char *query = getenv("QUERY_STRING");
    std::cout << "<html><body><p>" << query << "</p></body></html>";
    return 0;
}
